test/unit/metadata_test.py

```diff
@@ -39,3 +42,60 @@ def test_get_etl_group_tables() -> None:
     """Test that a Value error is raised for non existent etl group."""
     with pytest.raises(ValueError):
         Package.get_etl_group_tables("not_an_etl_group")
+
+
+@pytest.mark.parametrize("resource_name", sorted(PUDL_RESOURCES.keys()))
+def test_pyarrow_schemas(resource_name: str):
+    """Verify that we can produce pyarrow schemas for all defined Resources."""
+    _ = PUDL_RESOURCES[resource_name].to_pyarrow()
+
+
+@pytest.mark.parametrize("field_name", sorted(FIELD_METADATA.keys()))
+def test_field_definitions(field_name: str):
+    """Check that all defined fields are valid."""
+    _ = Field(name=field_name, **FIELD_METADATA[field_name])
```

Comment:
Would have caught those typos I mentioned above.

We can remove this after purging the unused fields.